lib/widgets/info_card.dart

```diff
@@ -18,85 +19,49 @@ class InfoCard<T> extends StatelessWidget {
-                            .textTheme
-                            .headline6
-                            .copyWith(fontSize: 18),
-                        overflow: TextOverflow.fade,
-                        maxLines: 2,
-                      ),
-                    ),
-                  if (onShowMore != null)
-                    GestureDetector(
-                      onTap: onShowMore,
-                      key: showMoreButtonKey ?? const ValueKey('show_more'),
-                      child: Row(
-                        children: <Widget>[
-                          Text(
-                            S.current.actionShowMore,
-                            style: Theme.of(context)
-                                .coloredTextTheme
-                                .subtitle2
-                                .copyWith(
-                                    color: Theme.of(context).primaryColor),
-                          ),
-                          Icon(
-                            Icons.arrow_forward_ios_outlined,
-                            color: Theme.of(context).primaryColor,
-                            size:
-                                Theme.of(context).textTheme.subtitle2.fontSize,
-                          )
-                        ],
-                      ),
-                    ),
-                ],
-              ),
-              const SizedBox(height: 10),
-              FutureBuilder(
-                  future: future,
-                  builder: (final context, final snapshot) {
-                    if (snapshot.connectionState == ConnectionState.done) {
-                      if (snapshot.hasData) {
-                        if ((snapshot.data is Map ||
-                                snapshot.data is Iterable) &&
-                            snapshot.data.isEmpty) {
-                          return noneYet(context);
-                        }
+    return FutureBuilder(
+        future: future,
+        builder: (final context, final snapshot) {
+          if (snapshot.connectionState == ConnectionState.done) {
+            // this condition resolves to true if the snapshot contains some data and if that data is iterable and not empty
+            final bool toDisplayContent = snapshot.hasData &&
+                !((snapshot.data is Map || snapshot.data is Iterable) &&
+                    snapshot.data.isEmpty);
 
-                        return builder(snapshot.data);
-                      } else {
-                        return noneYet(context);
-                      }
-                    }
-                    return const SizedBox(
-                      height: 100,
-                      child: Center(child: CircularProgressIndicator()),
-                    );
-                  }),
-            ],
-          ),
-        ),
-      ),
-    );
+            // if there is no content to be displayed, we default to the chosen strategy for the no-items widget
```

Comment:
Comments should be complete sentences (start with uppercase, end with a period). But I also don't feel this adds a lot of value, but rather uses terms that aren't used anywhere else, making it even more confusing - what strategy? What's the "no-items widget"? Try to imagine the person reading this comment has 0 context on what you're trying to do.